Tests/Fixtures/Form/ImprovedTestType.php

```diff
@@ -14,35 +25,74 @@ class ImprovedTestType extends AbstractType
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $choiceType = LegacyFormHelper::getType('Symfony\Component\Form\Extension\Core\Type\ChoiceType');
+        $datetimeType = LegacyFormHelper::getType('Symfony\Component\Form\Extension\Core\Type\DateTimeType');
+        $dateType = LegacyFormHelper::getType('Symfony\Component\Form\Extension\Core\Type\DateType');
+
         $builder
-            ->add('dt1', 'datetime', array('widget' => 'single_text', 'description' => 'A nice description'))
-            ->add('dt2', 'datetime', array('date_format' => 'M/d/y'))
-            ->add('dt3', 'datetime', array('widget' => 'single_text', 'format' => 'M/d/y H:i:s'))
-            ->add('dt4', 'datetime', array('date_format' => \IntlDateFormatter::MEDIUM))
-            ->add('dt5', 'datetime', array('format' => 'M/d/y H:i:s'))
-            ->add('d1', 'date', array('format' => \IntlDateFormatter::MEDIUM))
-            ->add('d2', 'date', array('format' => 'd-M-y'))
-            ->add('c1', 'choice', array('choices' => array('m' => 'Male', 'f' => 'Female')))
-            ->add('c2', 'choice', array('choices' => array('m' => 'Male', 'f' => 'Female'), 'multiple' => true))
-            ->add('c3', 'choice', array('choices' => array()))
-            ->add('c4', 'choice', array('choice_list' => new SimpleChoiceList(array('foo' => 'bar', 'bazgroup' => array('baz' => 'Buzz')))))
+            ->add('dt1', $datetimeType, array('widget' => 'single_text', 'description' => 'A nice description'))
```

Comment:
@bmxjama I took it from original repository, i would like not to change to our approach to give a easiest possiblillity to upgrade later.